cpp/src/io/parquet/experimental/hybrid_scan_helpers.cpp

```diff
@@ -529,4 +529,85 @@ aggregate_reader_metadata::filter_row_groups_with_bloom_filters(
   return bloom_filtered_row_groups.value_or(all_row_group_indices(row_group_indices));
 }
 
+/**
+ * @brief Converts column named expression to column index reference expression
+ */
+named_to_reference_converter::named_to_reference_converter(
+  std::optional<std::reference_wrapper<ast::expression const>> expr,
+  table_metadata const& metadata,
+  std::vector<SchemaElement> const& schema_tree)
+{
+  if (!expr.has_value()) { return; }
+  // create map for column name.
+  std::transform(metadata.schema_info.cbegin(),
+                 metadata.schema_info.cend(),
+                 thrust::counting_iterator<size_t>(0),
+                 std::inserter(_column_name_to_index, _column_name_to_index.end()),
+                 [](auto const& sch, auto index) { return std::make_pair(sch.name, index); });
+
+  auto const& root = schema_tree.front();
```

Comment:
Only change in the child class. Any `column_references` in the filter expression now need to first converted to names and then to new `column_references` (within the column selection)